rest/config_legacy.go

```diff
@@ -97,11 +100,43 @@ func (lc *LegacyServerConfig) ToStartupConfig() (*StartupConfig, DbConfigMap, er
 	// find a database's credentials for bootstrap (this isn't the first database config entry due to map iteration)
 	bsc := &BootstrapConfig{}
 	for _, dbConfig := range lc.Databases {
-		if dbConfig.Server == nil || *dbConfig.Server == "" {
+		server := dbConfig.Server
+		if server == nil || *server == "" {
 			continue
 		}
+
+		connSpec, err := gocbconnstr.Parse(*server)
+		if err != nil {
+			return nil, DbConfigMap{}, err
+		}
+
+		// support for username and password if not specified in username and password fields
+		splitServers := strings.Split(*server, ",")
+		// Only first URL is supported to have username and pass or else gocbconnstr.Parse does not work correctly
+		if strings.Contains(splitServers[0], "@") && dbConfig.Username == "" && dbConfig.Password == "" {
+			u, err := url.Parse(splitServers[0])
+			if err != nil {
+				return nil, DbConfigMap{}, err
+			}
+			username := u.User.Username()
+			password, passwordSet := u.User.Password()
+			if username != "" && passwordSet {
+				dbConfig.Username = username
+				dbConfig.Password = password
+			}
+		}
+
+		if connSpec.Scheme == "http" {
+			*server = "couchbase://"
+			for _, addr := range connSpec.Addresses {
+				*server = fmt.Sprintf("%s%s,", *server, addr.Host)
+			}
+			// Trailing , is valid but removed to make field neater
+			*server = strings.TrimSuffix(*server, ",")
+		}
```

Comment:
This whole block feels like it should be a separate function that can take a plain `gocbv1`/`http://` connection string and return a `gocbv2`/`couchbase://` compatible one.

It's almost the opposite of `CouchbaseURIToHttpURL` in `base/util.go`

Should make it easier to test more permutations of it too.